crypto/evp/skeymgmt_meth.c

```diff
@@ -116,8 +116,12 @@ static void *skeymgmt_from_algorithm(int name_id,
         ERR_raise(ERR_LIB_EVP, EVP_R_INVALID_PROVIDER_FUNCTIONS);
         return NULL;
     }
+
+    if (!ossl_provider_up_ref(prov)) {
+        EVP_SKEYMGMT_free(skeymgmt);
+        return NULL;
+    }
     skeymgmt->prov = prov;
-    ossl_provider_up_ref(prov);
```

Comment:
This already in https://github.com/openssl/openssl/pull/26831, please drop this.